### ISSUE TITLE
Release?

### ISSUE THREAD

**Issue by mrocklin** [dask/dask-mpi#15]
This seems usable?  Should we release?  

@kmpaul I'm happy to walk you through this if you like or just handle it for you.

**Comment by kmpaul** [comment]
I would love a walkthrough.  I've done releases on GitHub before (obviously), and I've published to PyPI.  I'm in the middle of trying to get a package published to conda-forge, but it has not been successful, yet.  So, if you can provide guidance on how this should be done for `dask-mpi`, that would be great!

**Comment by mrocklin** [comment]
Happy to screenshare sometime this morning.  I'm free most of the day.

In the meantime, here are the steps that I would do.

1.  Tag a release and push to github

        git tag -a 1.0.0 -m "Version 1.0.0"
        git push origin master --tags

2.  Build and publish release on PyPI

    ```
    git clean -xfd  # remove any files not checked into git
    python setup.py sdist bdist_wheel --universal  # build package
    twine upload dist/*  # register and push to pypi
    ```

3.  Build conda recipe
    
    fork github.com/conda-forge/staged-recipes
     ```
    git clone git@github.com:kmpaul/staged-recipes
    cd staged-recipes
    git checkout -b dask-mpi
    mkdir dask-mpi
    cp recipes/example/meta.yaml recipes/dask-mpi/meta.yaml
    # edit recipes/dask-mpi/meta.yaml
    ```

    -  set package name and version
    - get sha256 from pypi.org, similar to this page: https://pypi.org/project/dask/#files but for dask-mpi
    - uncomment `noarch: python`
    - remove c compiler build step (I think?)
    - fill in the rest of the information (should be simple, I think?)

    -  commit and submit a PR upstream

**Comment by kmpaul** [comment]
Ok.  I think that seems reasonable.  I've done most of this before.

...One question about the conda recipe, however.  Since `dask-mpi` depends on `MPI`...which I don't think is available on Windows, do I need to do anything special to indicate that this won't work on Windows?

**Comment by mrocklin** [comment]
I would put `mpi4py` in the requirements and then just be done with it.  I suspect that whatever they've done for `mpi4py` will transitively carry over to us.

**Comment by kmpaul** [comment]
Sounds reasonable.

**Comment by mrocklin** [comment]
I see that dask-mpi is up on PyPI.  Woot!

Offer still stands to screenshare for conda-forge.  I suspect that we could bang it out in fifteen minutes.

**Comment by kmpaul** [comment]
Yes!  😄  I'd like to bang away at the conda-forge step myself for a bit.  I need to learn this and get used to it.

**Comment by mrocklin** [comment]
Go for it :)

**Comment by kmpaul** [comment]
Incidentally, I followed a slightly different path when putting together the `setup.py` file.  I included in the repository `environment.yml` file(s), and not `requirements.txt` files.  I know that this is non-standard.  Would you like me to switch over to the `requirements.txt` file approach before finishing the release?  I can always modify, test, and then re-release on PyPI first, before the PR.

**Comment by kmpaul** [comment]
Actually, I can change that in the future, if desired (I'll make an issue for it).  I'll go ahead and submit the PR.

**Comment by mrocklin** [comment]
I think it's up to you.  I recommend doing whatever you find comfortable.
If an issue arises then it's easy enough to change.

**Comment by kmpaul** [comment]
Sounds good.  I'll wait to see if the PR goes through.

**Comment by mrocklin** [comment]
Looks like this is done.  

@kmpaul any objection if I tweet the following from the dask_dev account?

```
NCAR devs (@NCARKMPaul) just released dask-mpi, a package to make it easier to deploy Dask with mpirun or mpiexec on HPC systems

https://dask-mpi.readthedocs.io/en/latest/
```

I ask just because I mention NCAR explicitly (it's nice from my perspective to mention institutions) but I'd be happy to drop that if you'd prefer.  If you have alternative suggestions for text let me know.

**Comment by kmpaul** [comment]
No objection at all!  The mention of NCAR is fine. Thank you!
